### ISSUE TITLE
Make the fuzzy match in from_place select larger entities by default when there are multiple options

### ISSUE THREAD

**Issue by stuartlynn** [ljwolf/cenpy#54]
Currently the fuzzy match in from_place selects the first match from the _places table 

https://github.com/ljwolf/cenpy/blob/product/cenpy/products.py#L111

in some cases there can be multiple entries with the same name / state. In this case we want to select the largest of those place types instead of just a random one. The order of preference should be 

1. Incorporated Place
2. Census Designated Place
3. County Subdivision

This is discussed a little in #52

**Comment by ljwolf** [comment]
After investigating some, this can result in some somewhat surprising associations. It's not always clear that any of these geographies are "higher" or "larger" than others, so assuming so in the case of a tie isn't easy/simple. 

Instead, I think we should document this, improve the error text, and make sure that it's not guaranteed to work if you don't specify the place type in cases where there are many records with the same name of different place types. I don't even think a warning is appropriate, since that hits the most common use case...

**Comment by ljwolf** [comment]
I'll leave this to the future if there's a better way to do this than I've tried. but, basically, I've looked at comparing ties between the three types, and find that sometimes the hierarchy isn't respected at all. Further, it's not easy to judge programmatically how a match on type 1 compares to a match on type 3 if they're only slight matches to one another.

**Comment by stuartlynn** [comment]
Yeah that makes sense. Better to fail explicitly rather than doing something magic and opaque.
